common/fixation_entry.go

```diff
@@ -348,14 +408,113 @@ func (fs *FixationStore) PutEntry(ctx sdk.Context, index string, block uint64) {
 	fs.putEntry(ctx, entry)
 }
 
+// DelEntry deletes marks the entry for deletion so that future GetEntry will
+// not find it anymore. (Cleanup of the EntryIndex will be done when the last
+// reference is removed).
+func (fs *FixationStore) DelEntry(ctx sdk.Context, index string) error {
+	safeIndex, err := types.SanitizeIndex(index)
+	if err != nil {
+		return sdkerrors.ErrNotFound.Wrapf("invalid non-ascii index")
+	}
+
+	entry, found := fs.getUnmarshaledEntryForBlock(ctx, safeIndex, math.MaxUint64)
+	if !found || entry.Deleted {
+		return sdkerrors.ErrNotFound
+	}
+
+	block := uint64(ctx.BlockHeight()) + 1
+	if entry.Block >= block {
+		return sdkerrors.ErrInvalidHeight.Wrapf("unable to delete entry with future block: %d", block)
+	}
+
+	// add a marker version that is both deleted and stale to indicate this
+	// entry is deleted: "deleted" is needed to reject future GetEntry() calls;
+	// and "stale" is needed to reject future FindEntry() calls.
+
+	deleted := types.Entry{
```

Comment:
why not just mark the existing entry with deleted instead of adding a new entry?